designs/2019-core-options/README.md

```diff
@@ -0,0 +1,139 @@
+- Start Date: 2019-05-12
+- RFC PR: https://github.com/eslint/rfcs/pull/22
+- Authors: Toru Nagashima ([@mysticatea](https://github.com/mysticatea))
+
+# Configuring core options in Config Files
+
+## Summary
+
+This RFC adds several properties into config files to configure core options. People can use config files (including shareable configs!) instead of CLI options in order to configure some linter behavior.
+
+## Motivation
+
+We cannot configure some linter's behavior with config files, especially, shareable configs. It's convenient if we can configure those in shareable configs.
+
+## Detailed Design
+
+This RFC adds four properties to config files.
+
+```js
+// .eslintrc.js
+module.exports = {
+    disableInlineConfig: false,              // Corresponds to --no-inline-config
+    reportUnusedDisableDirectives: false,    // Corresponds to --report-unused-disable-directives
+    verifyOnRecoverableParsingErrors: false, // Corresponds to --verify-on-recoverable-parsing-errors
+    ignorePatterns: [],                      // Corresponds to --ignore-pattern
```

Comment:
I understand why this is changed from `ignorePattern` to `ignorePatterns` (and I do think it's a more accurate name given that it's an array), but like above, I wonder if the overhead of having to map CLI flags to slightly different config properties is worth having the name be slightly clearer.

Reply:
I'm not worried about this because some other options are similar. For example, `--global` ⇔ `globals: {}`, `--plugin` ⇔ `plugins: []`, and `--rule` ⇔ `rules: {}`.

Reply:
Due to the way optionator works, there is no way to do this without having some case look strange. This is because optionator allows you to specify multiple values in two ways: `--global a,b` or `--global a --global b`. Plural looks more natural for the first form and singular looks more natural for the second form.

I'm okay with CLI being different from CLIEngine in these cases if it makes sense.

Reply:
Makes sense!